### PR TITLE
Add FeaturePost support

### PR DESCRIPTION
**Pull request by makotech222**
See https://github.com/LemmyNet/lemmy/pull/2585

### CODE REVIEW

File: src/interfaces/source.ts [hunk 1]
```diff
@@ -286,14 +287,14 @@ export class ModLockPost {
...
-export class ModStickyPost {
+export class ModFeaturePost {
   id: number;
   mod_person_id: number;
   post_id: number;
   @Transform(({ value }) => toOption(value), { toClassOnly: true })
   @Transform(({ value }) => toUndefined(value), { toPlainOnly: true })
   @Expose()
-  stickied: Option<boolean>;
+  featured: Option<boolean>;
```
Comment (dessalines) [inline src/interfaces/source.ts:297]: Hrm.. this doesn't seem to make much sense for it to be optional. Better check the back end. Also I'll wait on merging cause the back-end still needs the local / community differentiation.

Comment (makotech222) [reply src/interfaces/source.ts:297]: I was thinking the same thing, wasn't sure why it was optional. Ill change it in that case.

Comment (dessalines) [reply src/interfaces/source.ts:297]: Once that's ready to go, I can merge and you should be able to fix the federation tests for https://github.com/LemmyNet/lemmy/pull/2585

Comment (makotech222) [reply src/interfaces/source.ts]: Done. Forgot to push it, loll.

**Review by dessalines — APPROVED** [review]
Looks good, thx.

**Comment by dessalines** [comment]
My bad, done: `0.17.0-rc.56`

**Pull request merged**